pkg/auth/migrate/rbac_to_acl.go

```diff
@@ -114,6 +118,35 @@ func RBACToACL(ctx context.Context, svc auth.Service, doUpdate bool, creationTim
 		messageFunc(group.DisplayName, *newACL, warnings)
 	}
 
+	users, _, err := svc.ListUsers(ctx, &model.PaginationParams{Amount: maxUsers + 1})
+	if err != nil {
+		return fmt.Errorf("list users: %w", err)
+	}
+	if len(users) > maxUsers {
+		return fmt.Errorf("%w (got %d)", ErrTooManyUsers, len(users))
+	}
+	for _, user := range users {
+		var warnings error
+		policies, _, err := svc.ListUserPolicies(ctx, user.Username, &model.PaginationParams{Amount: maxUserPolicies + 1})
+		if err != nil {
+			return fmt.Errorf("list users policies: %w", err)
+		}
+		if len(policies) > maxUsers {
+			return fmt.Errorf("%w (got %d)", ErrTooManyUsers, len(policies))
+		}
+
+		if len(policies) > 0 {
+			warnings = multierror.Append(warnings, fmt.Errorf("%w user %s - all policies will be detached from user", ErrPolicyAttachedToUsers, user.Username))
```

Comment:
Perhaps include the policies or their number?

Also consider
```suggestion
			warnings = multierror.Append(warnings, fmt.Errorf("%w user %s - detaching directly-attached policies", ErrPolicyAttachedToUsers, user.Username))
```
Or change text of ErrPolicyAttachedToUsers to create a nicer message here.